src/pack.c

```diff
@@ -1190,15 +1194,30 @@ int git_packfile_alloc(struct git_pack_file **pack_out, const char *path)
 	 * the index looks sane.
 	 */
 	if (git__suffixcmp(path, ".idx") == 0) {
-		size_t root_len = path_len - strlen(".idx");
-
+		root_len = path_len - strlen(".idx");
 		if (!git_disable_pack_keep_file_checks) {
 			memcpy(p->pack_name + root_len, ".keep", sizeof(".keep"));
 			if (git_path_exists(p->pack_name) == true)
 				p->pack_keep = 1;
 		}
 
 		memcpy(p->pack_name + root_len, ".pack", sizeof(".pack"));
+	} else {
+		root_len = path_len - strlen(".pack");
```

Comment:
In [line 1181](https://github.com/libgit2/libgit2/pull/5776/files#diff-226580d1635752a6f3386d44293260fbd4a33f2f405ba0b91fbd35c3d96315f9R1181) we ensure that `path_len` is at least `strlen(".idx")`.  If it were `strlen(".idx")`, here we would set `root_len` to `(size_t)-1`.

I think that it's unlikely that there's a call path that would call us with just `.idx` as the path but some sort of guard seems like it should be in order.